HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+    {
+        cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                cvMat.at<double>(i, j) = eigenMat(i, j);
+            }
+        }
+
+        return cvMat;
+    }
+
+    Eigen::MatrixXd cv2eigen(const cv::Mat &cvMat)
+    {
+        if(cvMat.dims > 2)
+        {
+            throw "Invalid matrix dimensions. Expected 2D.";
+        }
+
+        Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                eigenMat(i, j) = cvMat.at<double>(i, j);
+            }
+        }
+
+        return eigenMat;
+    }
+
+    Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+    {
+        Eigen::Vector4d reorderedAxisAngle;
+
+        for(int i = 0; i < 3; i++)
+            reorderedAxisAngle[i] = axisAngle.axis()[i];
+        reorderedAxisAngle[3] = axisAngle.angle();
+
+        return reorderedAxisAngle;
+    }
+
+    /** Converts Rotation Matrix to RollPitchYaw.
+     * The convention we use here is that Roll is a rotation about x-axis,
+     * Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+     * @param rotationMatrix Rotation Matrix to convert to RPY
+     * @param rotation Enum to indicate which convention to follow for this conversion
+     * @return rollPitchYaw array of angles, in radians
+     */
+    Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d rotationMatrix,
+                              const TransformUtils::Rotation rotation = ZYX_INTRINSIC)
+    {
+        Eigen::Array3d rollPitchYaw;
+        switch(rotation)
+        {
+            case TransformUtils::Rotation::XYZ_INTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+            case TransformUtils::Rotation::XYZ_EXTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+            case TransformUtils::Rotation::ZYX_INTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+            case TransformUtils::Rotation::ZYX_EXTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+            default: throw "Invalid rotation";
+        }
+        // The returned array should be ordered by Ox, Oy and Oz. Where Oa is the rotation about axis a.
+        switch(rotation)
+        {
+            case TransformUtils::Rotation::XYZ_EXTRINSIC: rollPitchYaw.reverseInPlace(); break;
+            case TransformUtils::Rotation::ZYX_INTRINSIC: rollPitchYaw.reverseInPlace(); break;
+        }
+
+        return rollPitchYaw;
+    }
+
+    /** Converts RollPitchYaw to Rotation Matrix.
+     * This function takes an array of roll, pitch and yaw angles, and a convention, as input parameters.
+     * For rollPitchYaw we define that \a roll is a rotation about x-axis,
```

Comment:
What's this `\a` syntax?

Reply:
Heard you were using Doxygen, so I thought I could use these [special commands](http://www.doxygen.nl/manual/commands.html#cmda).
You have another comment about extensive comments for two functions. Well, the conversion from Rotation Matrix to Roll-Pitch-Yaw is complicated, and requires solid definitions in order to work at all. Thus, without comments it's only guesswork involved in its usage.
Would you suggest just normal inline comments (//) inside the functions, instead of a style that is picked up by Doxygen (if we were to run that). I also found it useful that Visual Studio picks it up (IntelliSense?)

Reply:
I was just surprised to see Doxygen syntax in a project which doesn't use Doxygen, and doesn't use this syntax anywhere else.

The most important thing about these comments is that people reading them to understand the sample find them easy to understand. I guess in that sense, the `@param` and `@return` syntax can actually be useful even when we don't use Doxygen. So I don't mind if we use those. 

I don't know Doxygen well, but isn't the `\a` syntax meant for referring to a parameter name? So if you do `\a foo` it refers to the parameter `foo`? Since it doesn't do that here, I suggest removing the `\a`s. Also, for people unfamiliar with Doxygen (like me) I think the `@param` and `@return` syntax is self explanatory, but the `\a` stuff might just be confusing (it was for me).

I also don't think `@return` is supposed to mention the variable name it returns, it should just document what the return value is. So for instance for `rotmat2rpy` I think it should say just "array of angles, in radians".